src/core/reference/include/ngraph/runtime/reference/group_normalization.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/coordinate_transform.hpp"
+
+using namespace std;
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+
+template <typename T>
+void group_normalization(const T* const data,
+                         const T* const scale,
+                         const T* const bias,
+                         T* const out,
+                         const ov::Shape& data_shape,
+                         const size_t num_groups,
```

Comment:
Shouldn't the `num_groups` type (here size_t) be aligned with the type of `num_groups` (int64_t) from GroupNormalization core op?

Reply:
Overall the sign is validated during shape inference, so it's okay to keep unsigned in reference:
https://github.com/openvinotoolkit/openvino/blob/fc9e0ce315e77d8235b9c45917bf5cde125feecd/src/core/shape_inference/include/group_normalization_shape_inference.hpp#L22

Reply:
`size_t` is valid here and imo reference implementation should have a bit of independence ;)